pkg/logs/client/http/sync_destination.go

```diff
@@ -0,0 +1,108 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//nolint:revive // TODO(AML) Fix revive linter
+package http
+
+import (
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-agent/comp/logs/agent/config"
+	pkgconfigmodel "github.com/DataDog/datadog-agent/pkg/config/model"
+	"github.com/DataDog/datadog-agent/pkg/logs/client"
+	"github.com/DataDog/datadog-agent/pkg/logs/message"
+	"github.com/DataDog/datadog-agent/pkg/logs/metrics"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// SyncDestination sends a payload over HTTP and does not retry.
+type SyncDestination struct {
+	destination    *Destination
+	senderDoneChan chan *sync.WaitGroup
+}
+
+// NewSyncDestination returns a new synchronous Destination.
+func NewSyncDestination(endpoint config.Endpoint,
+	contentType string,
+	destinationsContext *client.DestinationsContext,
+	senderDoneChan chan *sync.WaitGroup,
+	telemetryName string,
+	cfg pkgconfigmodel.Reader) *SyncDestination {
+
+	return newSyncDestination(endpoint,
+		contentType,
+		destinationsContext,
+		time.Second*10,
+		senderDoneChan,
+		telemetryName,
+		cfg)
+}
+
+func newSyncDestination(endpoint config.Endpoint,
+	contentType string,
+	destinationsContext *client.DestinationsContext,
+	timeout time.Duration,
+	senderDoneChan chan *sync.WaitGroup,
+	telemetryName string,
+	cfg pkgconfigmodel.Reader) *SyncDestination {
+
+	return &SyncDestination{
+		destination:    newDestination(endpoint, contentType, destinationsContext, timeout, 1, false, telemetryName, cfg),
+		senderDoneChan: senderDoneChan,
+	}
+}
```

Comment:
Sorry, this is too verbose, is there a way to just compact this? What's the purpose of both functions if they are going to map to adding a new destination, can't we do it directly?

Reply:
It seem that's the only occurrence of the method, please lmk what you think

Reply:
Ah yeah, my bad -- this is almost an exact copy of `client/http/destination.go` but this sub method isn't necessary here.

Reply:
I'd ask someone from whichever team is responsible for their thoughts on removing it, but up to you